### PR TITLE
perf(ai): clarify coach tool routing descriptions

### PR DESCRIPTION
**Pull request by JeffOtano**
## Summary

- Standardizes all registered coach tool descriptions with explicit use, non-use, and input/output guidance.
- Adds AI SDK `inputExamples` for the complex `program_week`, `create_workout`, `add_exercise`, and `set_goal` tool inputs.
- Trims redundant prompt-level routing text now covered by the tool descriptions.

Refs #189.

## Changes

- Tool definitions: clarified routing boundaries across general data tools, weekly programming tools, draft modification tools, and coaching-domain tools.
- Complex examples: added compact `inputExamples` for the four high-ambiguity tools called out in the issue.
- Prompt: reduced `TOOL USAGE` to a capability map so detailed selection rules live with the tool definitions.
- Tests: added `convex/ai/toolDescriptions.test.ts` to enforce description coverage and required input examples.

## Checklist

- [x] `npx tsc --noEmit` passes
- [x] `npm run lint` passes
- [x] `npm test` passes (existing + new tests)
- [x] New logic has tests (happy path + at least one error/edge case)
- [ ] No new or materially expanded file exceeds the 300-line soft cap
- [x] No file exceeds the 400-line hard cap (enforced by CI)
- [x] Functions stay near the 60-line convention
- [x] No new `any` types (enforced by ESLint); `as` casts only at deserialization boundaries
- [x] Tested in browser (N/A - no UI changes)
- [x] Commits follow conventional format (`type: description`)
- [x] No unused exports or dead code introduced

## Test Plan

- `npx tsc --noEmit`
- `npm run typecheck`
- `npm run lint`
- `npm run format:check`
- `npx vitest run convex/ai/toolDescriptions.test.ts convex/ai/promptSections.test.ts convex/ai/tools.test.ts convex/ai/weekModificationTools.test.ts`
- `npx vitest run convex/ai/coachEvals.test.ts` (skipped: no `GOOGLE_GENERATIVE_AI_API_KEY` available in the worktree or root env)
- `npm test`

Note: `convex/ai/tools.ts`, `convex/ai/weekModificationTools.ts`, and `convex/ai/coachingTools.ts` were already above the 300-line soft cap before this change. This PR keeps them below the 400-line hard cap and avoids a file split because the change is tool-definition copy plus examples, not a new runtime abstraction.

## Summary by CodeRabbit

* **Chores**
  * Expanded and clarified descriptions for many AI coaching tools (planning, weekly programs, duration estimation, exercise/week modification, goals, injuries, and workouts) and added input examples for several tools to improve usage guidance and output consistency.
  * Rewrote high-level tool usage guidance for the coaching agent.

* **Tests**
  * Added metadata validation tests to enforce non-empty descriptions and presence/format of input examples for key tools.

[Review Change Stack](https://app.coderabbit.ai/change-stack/JeffOtano/roni/pull/371)

### CODE REVIEW

File: convex/ai/toolDescriptions.test.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const REQUIRED_EXAMPLE_TOOLS = [
+  "program_week",
+  "create_workout",
+  "add_exercise",
+  "set_goal",
+] as const satisfies (keyof typeof coachAgentConfig.tools)[];
+
+function asToolDefinition(tool: unknown): ToolDefinition {
+  return tool && typeof tool === "object" ? (tool as ToolDefinition) : {};
+}
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/ai/toolDescriptions.test.ts:22]: Good catch. I replaced the unchecked cast with a Zod `safeParse` path so `inputExamples` must actually be an array before the tests inspect it. Verified with `npx tsc --noEmit`, `npm run lint`, and `npm test`. Resolving this thread.

File: convex/ai/toolDescriptions.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  it("keeps routing guidance on every registered tool", () => {
+    for (const { name, tool } of tools) {
+      expect(tool.description, `${name} is missing a description`).toBeTruthy();
+      expect(tool.description, `${name} should say when to use it`).toContain("Use");
+      expect(tool.description, `${name} should say when not to use it`).toMatch(
+        /Do not|Never|Does not/,
+      );
+      expect(tool.description, `${name} should describe inputs and outputs`).toMatch(
+        /Inputs?.*returns?|returns?.*Inputs?/i,
+      );
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/ai/toolDescriptions.test.ts]: Good catch. I updated the matcher to use `[\\s\\S]*`, so the input/output assertion covers multiline descriptions too. Verified with `npx tsc --noEmit`, `npm run lint`, and `npm test`. Resolving this thread.

File: convex/ai/toolDescriptions.test.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+describe("coach tool descriptions", () => {
+  const tools = Object.entries(coachAgentConfig.tools).map(([name, tool]) => ({
+    name,
+    tool: asToolDefinition(tool),
+  }));
+
+  it("keeps routing guidance on every registered tool", () => {
+    for (const { name, tool } of tools) {
+      expect(tool.description, `${name} is missing a description`).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/ai/toolDescriptions.test.ts]: Good catch. I tightened this test to trim the tool description before checking length, then reuse that trimmed value for the routing-guidance assertions. Verified with `npx tsc --noEmit`, `npx vitest run convex/ai/toolDescriptions.test.ts`, `npm run format:check`, `npm run lint`, and `npm test`. Resolving this thread.
